.github/workflows/java-client-v4-integration-tests.yml

```diff
@@ -6,6 +6,10 @@ on:
     types:
       - completed
 
+# allow this workflow to update the status of the PR that triggered it
+permissions:
+  statuses: write
```

Comment:
Apart from the documentation I'm making this change to fix a `403` response when trying to update the PR status https://github.com/conductor-oss/conductor/actions/runs/10930113251/job/30342332639#step:7:29